_posts/2020-03-04-#16981.md

```diff
@@ -12,11 +12,38 @@ commit:
 
 ## Notes
 
-_Notes and questions to follow!_
+- The `reindex` command wipes and rebuilds the two leveldb databases for the block index and the chainstate, using the block files saved on disk. This procedure is completely local, it requires no interactions with the p2p network.
+
+- Reindexing is a lengthy procedure taking several hours on mainnet and consists of two steps:
+
+    1. Rebuilding the index of blocks based on the files `blk00???.dat` files saved in `.bitcoin/blocks`.
+
+    2. Rebuilding the chainstate (UTXO set) by fully validating each block starting from genesis, using the block index created in step 1.
+
+- There is a second command, `reindex-chainstate` that will only perform step 2. This PR improves the runtime of step 1 and does not affect step 2 (which is already highly optimized since it uses the same validation functions that are used for connecting new blocks received from the network).
+
+- Reindex uses the `CBufferedFile` stream, introduced in [#1962](https://github.com/bitcoin/bitcoin/pull/1962), which has a buffer to allow "rewinding" the stream position to an earlier position without additional disk access. In the merged PR [#16577](https://github.com/bitcoin/bitcoin/pull/16577), the author of this PR fixed a bug in  `CBufferedFile` and added comprehensive unit tests.
```

Comment:
I think 'of this PR' refers to the author of the PR we're reviewing this week, so I think it makes sense to keep this.

Reply:
yes, will change to "this week's PR" to make it even more clear.

Reply:
thanks -- yes, I was confused here

Reply:
Re summarising the bug - I wouldn't go into too much detail here beyond giving the link to the PR, because it was a rather technical issue with the implementation of `CBufferedFile` and not directly related to the changes of this PR.